trigger_circleci_pipeline.sh

```diff
@@ -7,38 +7,42 @@ CIRCLECI_PROJECT_API_BASE="${CIRCLECI_API_BASE}/project/github/${GITHUB_REPOSITO
 CIRCLE_WORKFLOW_URL_BASE="https://circleci.com/workflow-run"
 
 branch="${BRANCH_REF#refs/heads/}"
-workflow_to_trigger=$1
-triggered_key=${2:-""}
+
+request_data=$(
+    jq \
+        --arg triggered_workflow "${TRIGGERED_WORKFLOW}" \
+        --arg branch "${branch}" \
+        '{"branch": $branch, "parameters": (. += {"triggered_workflow": $triggered_workflow})}' <<< "${WORKFLOW_KWARGS}"
+)
 
 trigger_pipeline_response=$(
     curl -sSX POST \
         -H 'Content-type: application/json' \
         -H "Circle-Token: ${CIRCLECI_USER_PERSONAL_API_TOKEN}" \
         -H "x-attribution-login: ${GITHUB_ACTOR}" \
         -H "x-attribution-actor-id: medigate-ci-functions" \
-        -d "{
-          \"branch\": \"${branch}\",
-          \"parameters\": {
-            \"triggered_workflow\": \"${workflow_to_trigger}\",
-            \"triggered_key\": \"${triggered_key}\"
-          }
-        }" \
+        -d "${request_data}" \
         "${CIRCLECI_PROJECT_API_BASE}/pipeline"
 )
 
+if [ ! $(jq 'has("triggered_workflow")' <<< "${WORKFLOW_KWARGS}") = false ]; then
```

Comment:
I would move this to line 10, before we send any data to circle ci, there is no reason to call the circle ci API if the data is invalid

Reply:
For some reason I read the code earlier as if `trigger_pipeline_response` is a function; this remains from before I noticed otherwise :sweat_smile: